src/libraries/Common/tests/System/TimeProviderTests.cs

```diff
@@ -174,10 +174,15 @@ public void FastClockTest()
             Assert.Equal(stamp2 - stamp1, fastClock.GetElapsedTime(stamp1, stamp2).Ticks);
         }
 
+        public class DerivedTimeProvider : TimeProvider
```

Comment:
is this sufficient to protect the fix?

Reply:
Indeed, this approach will allow us to test the code path where `timePovider != TimeProvider.System` and cover the test cases that reproduce the existing issue, such as creating a CancellationTokenSource with an infinite duration.